python/rikai/video/ffmpeg.py

```diff
@@ -0,0 +1,116 @@
+#  Copyright 2021 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""FFmpeg related types and utils"""
+
+import sys
+import numpy as np
+
+try:
+    import ffmpeg
+except ImportError:
+    raise ValueError(
+        "Couldn't import ffmpeg. Please make sure to "
+        "`pip install ffmpeg-python` explicitly or install "
+        "the correct extras like `pip install rikai[all]`"
+    )
+from rikai.types.vision import Image
+from rikai.types.video import VideoSampler
+
+
+class VideoFrameSampler(VideoSampler):
```

Comment:
Does this replace SingleFrameSampler using OpenCV? Seems like these are duplicates?

Reply:
Modeled this from `SingleFrameSampler` in case we wanted to load all frames or iterate.